base/sparse/sparsematrix.jl

```diff
@@ -1033,6 +1104,63 @@ function getindex_I_sorted{Tv,Ti}(A::SparseMatrixCSC{Tv,Ti}, I::AbstractVector,
     return SparseMatrixCSC(nI, nJ, colptrS, rowvalS, nzvalS)
 end
 
+function getindex_I_sorted_binary_I{Tv,Ti}(A::SparseMatrixCSC{Tv,Ti}, I::AbstractVector, J::AbstractVector)
+    const nI = length(I)
+    const nJ = length(J)
+
+    colptrA = A.colptr; rowvalA = A.rowval; nzvalA = A.nzval
+    colptrS = Array(Ti, nJ+1)
+    colptrS[1] = 1
+
+    m = A.m
+    cacheI = zeros(Int, m)
+    # count rows
+    @inbounds for j = 1:nJ
+        col = J[j]
+        @simd for ptrA in colptrA[col]:(colptrA[col+1]-1)
+            cacheI[rowvalA[ptrA]] += 1
+        end
+    end
+
```

Comment:
Since `J` would be a subset of the columns, don't we need to step through `J` to be able to count `ptrS` correctly?